actioncable/lib/action_cable/channel/streams_handlers/custom.rb

```diff
@@ -0,0 +1,29 @@
+module ActionCable
+  module Channel
+    module StreamsHandlers
+      class NullCoder
+        class << self
+          def decode(val)
+            val
+          end
+
+          alias encode decode
+        end
+      end
+
+      class Custom < Base # :nodoc:
+        attr_reader :handler, :coder
+
+        def initialize(channel, handler:, coder:)
+          super
+          @handler = handler
+          @coder = coder || NullCoder
+        end
+
+        def call(message)
+          handler.(coder.decode(message))
+        end
+      end
```

Comment:
This isn't inheriting anything substantial from Base—the inheritance relationship is misleading.